sunpy/net/hek/utils.py

```diff
@@ -0,0 +1,151 @@
+            table = parse_unit(table, attribute, is_coord_prop)
+            unit_attr = ""
+            if is_coord_prop:
+                unit_attr = "event_coordunit"
+            else:
+                unit_attr = attribute["unit_prop"]
+
+            new_column = []
+            for idx, value in enumerate(table[attribute["name"]]):
+                if value in ["", None]:
+                    new_column.append(value)
+                elif attribute.get("is_chaincode", False):
+                    new_column.append(parse_chaincode(value, idx, attribute, table[attribute["unit_prop"]][idx]))
+                else:
+                    new_column.append(value * get_unit(attribute["unit_prop"], table[unit_attr][idx], is_coord_prop = is_coord_prop))
+            table[attribute["name"]] = new_column
+
+    for attribute in attributes:
+        if attribute.get("is_unit_prop", False) and attribute["name"] in table.colnames:
+            del table[attribute["name"]]
+    return table
+
+# NOTE: Needs unit test
+def parse_unit(table, attribute, is_coord_prop = False):
+    if attribute.get("is_chaincode", False):
+        return table
+    unit_attr = ""
+    if is_coord_prop:
+        unit_attr = "event_coordunit"
+    else:
+        unit_attr = attribute["unit_prop"]
+    for row in table:
+        if unit_attr in table.colnames and row[unit_attr] not in ["", None] and table[attribute["name"]].unit is not None:
+            table[attribute["name"]].unit = get_unit(attribute["unit_prop"], row[unit_attr], is_coord_prop = is_coord_prop)
+            break
+    return table
+
+# NOTE: Needs unit test
+def parse_chaincode(value, idx, attribute, unit_prop):
+    coord1_unit = u.deg
+    coord2_unit = u.deg
+    if attribute["frame"] == "helioprojective":
+        coord1_unit = u.arcsec
+        coord2_unit = u.arcsec
+    elif attribute["frame"] == "heliocentric":
+        coord1_unit = u.R_sun
+        coord2_unit = u.deg
+    elif attribute["frame"] == "icrs":
+        coord1_unit = get_unit("coord1_unit", unit_prop, is_coord_prop = True)
+        coord2_unit = get_unit("coord2_unit", unit_prop, is_coord_prop = True)
+
+    coordinates_str = value.split('((')[1].split('))')[0]
+    coord1_list = [float(coord.split()[0]) for coord in coordinates_str.split(',')] * coord1_unit
+    coord2_list = [float(coord.split()[1]) for coord in coordinates_str.split(',')] * coord2_unit
+    vertices = {}
+    if attribute["frame"] == "heliocentric":
+       vertices = SkyCoord(coord1_list, coord2_list, [1]* len(coord1_list)* u.AU, representation_type = "cylindrical", frame = "heliocentric")
+    else:
+        vertices = SkyCoord(coord1_list, coord2_list, frame=attribute["frame"])
+
+    return PolygonSkyRegion(vertices = vertices)
+
+# NOTE: Needs unit test
+def get_unit(unit_prop, str, is_coord_prop = False):
+    """Converts str into astropy unit."""
```

Comment:
Ideally the doc strings will follow the numpydoc style: https://numpydoc.readthedocs.io/en/latest/format.html 

This involves adding a description, a parameters section.